File: Project/Utils/OrganismFactory.java
package Utils;

import org.json.JSONObject;
import java.util.InvalidPropertiesFormatException;

import Simulator.World;
import Simulator.Organisms.Organism;
import Simulator.Organisms.Animals.*;
import Simulator.Organisms.Plants.*;

import Simulator.Organisms.Organism.Type;

public class OrganismFactory {

    public Organism create(JSONObject json, World world) throws InvalidPropertiesFormatException {

        Type type = Organism.Type.getTypeByInt(json.getInt("type"));

        switch (type) {
            case ANTELOPE:
                return new Antelope(json, world);
            case FOX:
                return new Fox(json, world);
            case HUMAN:
                return new Human(json, world);
            case CYBER_SHEEP:
                return new CyberSheep(json, world);
            case SHEEP:
                return new Sheep(json, world);
            case TURTLE:
                return new Turtle(json, world);
            case WOLF:
                return new Wolf(json, world);
            case GRASS:
                return new Grass(json, world);
            case GUARANA:
                return new Guarana(json, world);
            case MILKWEED:
                return new Milkweed(json, world);
            case SOSNOWSKY_HOGWEED:
                return new SosnowskyHogweed(json, world);
            case WOLF_BERRIES:
                return new WolfBerries(json, world);
            default:
                throw new InvalidPropertiesFormatException("Invalid type");
        }
    }
}

File: Project/Utils/OrganismFactory.py
from Simulator.Organisms.Organism import Type
from Simulator.Organisms.Animals import *
from Simulator.Organisms.Plants import *


class OrganismFactory:
    def __init__(self, world):
        self.world = world

    def create(self, json):
        type = Type.getTypeByInt(json["type"])
        if type == Type.ANTELOPE:
            return Antelope(self.world, json)
        elif type == Type.CYBER_SHEEP:
            return CyberSheep(self.world, json)
        elif type == Type.FOX:
            return Fox(self.world, json)
        elif type == Type.HUMAN:
            return Human(self.world, json)
        elif type == Type.SHEEP:
            return Sheep(self.world, json)
        elif type == Type.TURTLE:
            return Turtle(self.world, json)
        elif type == Type.WOLF:
            return Wolf(self.world, json)
        elif type == Type.GRASS:
            return Grass(self.world, json)
        elif type == Type.GUARANA:
            return Guarana(self.world, json)
        elif type == Type.MILKWEED:
            return Milkweed(self.world, json)
        elif type == Type.SOSNOWSKY_HOGWEED:
            return SosnowskyHogweed(self.world, json)
        elif type == Type.WOLF_BERRIES:
            return WolfBerries(self.world, json)
        else:
            raise Exception("Invalid type")

    def createFromType(self, type):
        if type == Type.ANTELOPE:
            return Antelope(self.world)
        elif type == Type.CYBER_SHEEP:
            return CyberSheep(self.world)
        elif type == Type.FOX:
            return Fox(self.world)
        elif type == Type.HUMAN:
            return Human(self.world)
        elif type == Type.SHEEP:
            return Sheep(self.world)
        elif type == Type.TURTLE:
            return Turtle(self.world)
        elif type == Type.WOLF:
            return Wolf(self.world)
        elif type == Type.GRASS:
            return Grass(self.world)
        elif type == Type.GUARANA:
            return Guarana(self.world)
        elif type == Type.MILKWEED:
            return Milkweed(self.world)
        elif type == Type.SOSNOWSKY_HOGWEED:
            return SosnowskyHogweed(self.world)
        elif type == Type.WOLF_BERRIES:
            return WolfBerries(self.world)
        else:
            raise Exception("Invalid type")
